pkg/agent/ssh/server.go

```diff
@@ -285,3 +324,62 @@ func (s *Server) sftpHandler(session ssh.Session) {
 		fmt.Println("sftp server completed with error:", err)
 	}
 }
+
+func (s *Server) getUserUidGid(username string) (int, int, []int, error) {
+	cmd := exec.Command("id", "-u", username)
+	uidBytes, err := cmd.Output()
+	if err != nil {
+		return 0, 0, nil, err
+	}
+
+	cmd = exec.Command("id", "-g", username)
+	gidBytes, err := cmd.Output()
+	if err != nil {
+		return 0, 0, nil, err
+	}
+
+	uid, err := strconv.Atoi(strings.TrimSpace(string(uidBytes)))
+	if err != nil {
+		return 0, 0, nil, err
+	}
+	gid, err := strconv.Atoi(strings.TrimSpace(string(gidBytes)))
+	if err != nil {
+		return 0, 0, nil, err
+	}
+
+	//	todo: get all user groups
+
+	dockerGroup, err := user.LookupGroup("docker")
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Avoid using `Fatal` anywhere outside the entrypoint. Return an error instead.

Also, this will error if the user is not part of the `docker` group or if the group doesn't exist at all.
That's really not ideal. The groups should just be empty in that case.